### Beginning of the Assignment - exploration

In [3]:
import pandas as pd
import numpy as np
import seaborn as sns
#pd.set_option('max_columns', 200)

In [42]:
character_nicknames_df = pd.read_csv('datasets/PHIL DATA SET/character_nicknames.csv')

In [28]:
details_df = pd.read_csv('datasets/PHIL DATA SET/details.csv')

In [29]:
favs_df = pd.read_csv('datasets/PHIL DATA SET/favs.csv')


In [30]:
person_alternate_names_df = pd.read_csv('datasets/PHIL DATA SET/person_alternate_names.csv')


In [31]:
person_anime_works_df = pd.read_csv('datasets/PHIL DATA SET/person_anime_works.csv')


In [32]:
person_details_df = pd.read_csv('datasets/PHIL DATA SET/person_details.csv')


In [33]:
stats_df = pd.read_csv('datasets/PHIL DATA SET/stats.csv')


##### First look

In [43]:
character_nicknames_df.shape

(37080, 2)

In [44]:
character_nicknames_df.head()

character_mal_id              nickname
0            280205              Hikaruko
1            280129              Hinacchi
2            280127         Bertha Willis
3            280066                 Jimmy
4            280059  Full Body Red Square

In [45]:
character_nicknames_df.columns
#will list all the columns. Not necessary here but kept as a pattern to follow with the following files

Index(['character_mal_id', 'nickname'], dtype='object')

In [46]:
character_nicknames_df.dtypes

character_mal_id     int64
nickname            object
dtype: object

In [47]:
character_nicknames_df.describe()
#not needed here because it should just be ids

character_mal_id
count      37080.000000
mean      115767.769579
std        87596.063426
min            3.000000
25%        30541.750000
50%       109656.000000
75%       192520.750000
max       282255.000000

### Data preparation (cleaning)


##### On the first dataset we may need to check for duplicates.
What does that mean? We have 102 rows that are duplicated over a 37080 rows dataset.
Why is that? Are there some characters that have multiple nicknames, so they are repeated in the dataset?

In [48]:
character_nicknames_df.loc[df.duplicated()]
#by default will give us the second

character_mal_id        nickname
328              276501            Koko
329              276501         Cao Cao
352              275883            King
356              275835  Eldest Brother
447              274437   The Half-Fool
...                 ...             ...
37051            281206          Apemon
37052            281206          Apemon
37053            281206          Apemon
37054            281206          Apemon
37068            281717         Colosso

[102 rows x 2 columns]

In [49]:
character_nicknames_df.query('character_mal_id == 75628')

character_mal_id        nickname
21558             75628        Anneroze
21559             75628  Witch of Steel
21560             75628      Iron Witch
21561             75628        Anneroze
21562             75628  Witch of Steel
21563             75628      Iron Witch

Mhh they're somehow different so yeah, the same character could have different nicknames.
We want to drop though the ones that are exactly the same.

In [50]:
#this way we drop the duplicates on the first dataset

character_nicknames_df = character_nicknames_df\
    .loc[~character_nicknames_df.duplicated()]\
    .reset_index(drop=True).copy()

#we don't need to use a subset here because there are just 2 columns

##### On the second dataset we may need to check for missing values and/or inconsistent values, since there are no duplicates

In [51]:
details_df.isna().sum()

character_mal_id     0
nickname            16
dtype: int64

In [1]:
### CIAOOOOOO